filter-core/src/main/java/org/wso2/micro/gateway/filter/core/api/RestAPI.java

```diff
@@ -85,63 +74,22 @@ public APIConfig getAPIConfig() {
-                resourceConfig.setPath(name);
-                resourceConfig.setMethod(ResourceConfig.HttpMethods.valueOf(httpMethod.name()));
-                resourceConfigs.add(resourceConfig);
-            });
-        });
-        return resourceConfigs;
-    }
-
-    private String resolveBasePath() {
-        String basePath = "/";
-        if (openAPI.getExtensions() != null && openAPI.getExtensions().containsKey(APIConstants.X_WSO2_BASE_PATH)) {
-            return openAPI.getExtensions().get(APIConstants.X_WSO2_BASE_PATH).toString();
-        }
-        List<Server> servers = openAPI.getServers();
-        if (servers != null) {
-            Server server = servers.get(0);
-            String url = replaceOpenAPIServerTemplate(server);
-            try {
-                basePath = new URI(url).getPath();
-            } catch (URISyntaxException e) {
-                logger.error(e);
-            }
-        }
-        return basePath;
-    }
-
     /**
-     * Open API server object can have server templating. Ex: https://{customerId}.saas-app.com:{port}/v2.
-     * When adding the back end url this method will replace the template values with the default value.
+     * Extract elected resource details from the request attributes.
      *
-     * @param server {@link Server} object of the open API definition
-     * @return templated server url replaced with default values
+     * @param attributes request attributes
+     * @return resource configuration identified by the request
      */
-    private static String replaceOpenAPIServerTemplate(Server server) {
-        //server url templating can have urls similar to 'https://{customerId}.saas-app.com:{port}/v2'
-        String url = server.getUrl();
-        Pattern serverTemplate = Pattern.compile("\\{.*?}");
-        Matcher matcher = serverTemplate.matcher(url);
-        while (matcher.find()) {
-            if (server.getVariables() != null && server.getVariables()
-                    .containsKey(matcher.group(0).substring(1, matcher.group(0).length() - 1))) {
-                String defaultValue = server.getVariables()
-                        .get(matcher.group(0).substring(1, matcher.group(0).length() - 1)).getDefault();
-                url = url.replaceAll("/" + matcher.group(0), defaultValue);
-            } else {
-                logger.error("Open API server url templating is used for the url : " + url
-                        + ". But default values is not provided for the variable '" + matcher.group(0)
-                        + "'. Hence correct url will not be resolved during the runtime "
-                        + "unless url is overridden during the runtime");
-            }
-        }
-        return url;
+    private List<ResourceConfig> extractResourceConfig(Map<String, String> attributes) {
+        String resPath = attributes.get(APIConstants.GW_RES_PATH_PARAM);
+        String resMethod = attributes.get(APIConstants.GW_RES_METHOD_PARAM);
```

Comment:
If multiple http methods are involved with a single path, the filter-core gets a space separated set of Methods (in a pending PR).
Hence shall we split this using space and add a resource for each method.

Reply:
Fixed with c2b0467